packages/update-monitor/src/discovery/DiscoveryEngine.ts

```diff
@@ -9,7 +9,7 @@ import { DiscoveryOptions } from './DiscoveryOptions'
 
 export class DiscoveryEngine {
   constructor(
-    private readonly provider: providers.Provider,
+    readonly provider: providers.Provider,
```

Comment:
Why?

Reply:
Related to your previous comment. During discovery(!), we need `provider` to call a contract that is not verified (a proxy that returns implementation, so we need to get it somehow). That's why I had to expose provider from DiscoverEngine which is the only object discover process receives.